### PR TITLE
Fix RC between publishers with duplicated ids

### PR DESCRIPTION
**Pull request by mickel8**
It might happen that the old live view is still alive when the new one is created - e.g. when a page refresh happens, the backend process is alive for over 10 seconds (maybe it's waiting for reconnects?). 

This might result in two processes publishing on the same topic. This PR fixes this issues by introducing `locked` flag that is released when we stop getting tracks info from the old publisher or we get the bye message.

Credits for the solution @brzep :clap:

### CODE REVIEW

File: lib/live_ex_webrtc/player.ex [hunk 1]
```diff
@@ -80,6 +80,9 @@ defmodule LiveExWebRTC.Player do
 
   @type t() :: struct()
 
+  @check_lock_timeout_ms 3000
+  @max_lock_timeout 3000
```
Comment (sgfn) [inline lib/live_ex_webrtc/player.ex:84]: for consistency
```suggestion
  @max_lock_timeout_ms 3000
```

File: lib/live_ex_webrtc/player.ex [hunk 2]
```diff
@@ -427,6 +474,27 @@ defmodule LiveExWebRTC.Player do
...
+      {:noreply, socket}
+    end
+  end
+
+  @impl true
+  def handle_info(:check_lock, socket) do
+    socket = assign(socket, :lock_timer, nil)
+    {:noreply, socket}
+  end
+
```
Comment (sgfn) [inline lib/live_ex_webrtc/player.ex:497]: I feel like we should be using `socket.assigns.player.foobar` rather than `socket.assigns.foobar` here as well

**Review by sgfn — CHANGES_REQUESTED** [review]

**Comment by mickel8** [comment]
@sgfn I revisited how publisher is cleaned up. Could you test your edge cases and see if everything works correctly?

**Comment by sgfn** [comment]
@mickel8 sure I can, just be warned that this has the potential to further delay the demo

At first glance I'm worried certain things will happen multiple times...
